src/common/reducers/entities/index.js

```diff
@@ -1,27 +1,43 @@
 import merge from 'lodash/merge';
 
 import * as RepoActionTypes from '../../actions/repository';
-
 import repository from './repository';
 
-
 // TODO snaps
 
-export function entities(state = { snaps: {}, repos: {} }, action) {
+export function entities(state = {
+  snaps: {},
+  repos: {},
+  owners: {}
+}, action) {
   if (action.payload && action.payload.entities) {
     return merge({}, state, action.payload.entities);
   }
 
   // only modify repos if action is one of REPO_ types
-  if (action.payload && action.payload.id && RepoActionTypes[action.type]) {
+  if (RepoActionTypes[action.type]) {
```

Comment:
I'm inclined to keep this as it is for now, and come back to it on the snaps refactor. The return  from your example in the case where state is undefined would be a new state object, I'm not 100% sure if that's OK to do.

I'm also not sure we even need the check on RepoActionType, it seems like a sane thing to do to me but if it's not there state just passes through untouched ... and maybe that's better.

But, given deadlines, I suggest merging and dealing with this later?